reana_server/scheduler.py

```diff
@@ -234,14 +239,13 @@ def on_message(self, body, message):
                 )
                 self._fail_workflow(workflow_id, logs=error_message)
             except Exception as e:
-                logging.error(
-                    f"Something went wrong while calling RWC:\n {e}", exc_info=True
-                )
+                error = f"Something went wrong while calling reana-workflow-controller:\n{e}"
```

Comment:
Should we expose this to the user, including the exception message?

Reply:
It would be good to expose this information as well, however all the whlist making sure that nothing sensitive can ever leak out that way, such as DB host information and such. (Thinking about the second exception catching in this PR.)

To mitigate such a risk, we could log two things, (i) the generic error that we would expose to the user (_"Something went wrong while calling reana-workflow-controller."_) and (ii) detailed error including exception detail that would go into the pod log only (for the admins).

For example, if the error looks like:

```
HTTPConnectionPool(host='reana-workflow-controller.default.svc.cluster.local', port=80): Max retries exceeded with url: /api/workflows/myanalysis/workspace?user=00000000-0000-0000-0000-000000000000&file_name=%2Fspecs%2Fsteps.yml (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f171cc0a400>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))
```

then the user may not be able to act much on such an error anyway.

(That said, if we are fully sure that nothing sensitive could ever leak out via that traceback exception, then we could consider exposing also the exception error to the user, possibly after passing it via massaging function that could strip away any unwanted bits. However it seems better to stay on the more prudent side.)

Reply:
Implemented option (i)